### PR TITLE
Create custom block type uiowa_events for use in layout builder.

### LINKED ISSUE #2071: Create custom events block for use in Layout Builder
There are quite a few different options here. Considering there is already an event block built out, it could be confusing to have a separate events block for content hub integration if we plan on making the static event block generally available. Additionally, there has been talk of sites needing an event content type which further complicates things.

# Potential options
1. Create a separate event list block that pulls events from content hub.
  a. Keep the static event block in a split.
  b. Make the static event block generally available.
2. Integrate the event list block into the static block and allow the user to choose.
  a. This could also include a single event reference from content hub which would make for three choices.
3. An event content type could present a fourth option if integrated into this all-encompassing event block.

MVP, and probably the path of least resistance, is probably option 1 while keeping the static block in a spilt. However, I think the static event option is nice to have and offers more flexibility - we will eventually have requests for one-off events that are not on content hub for some reason. Thats why I prefer option 2 but have concerns of future integration of option 3, mainly over how intuitive and usable the interface would be. It could be done though with the states API and is at least accurately representing the myriad ways of creating/referencing events on campus.

### PR DESCRIPTION
**Pull request by richardbporter**
Resolves #2071 

### Testing
- Sync a site via `blt sync`.
- Add an Events block to a page. Try different combinations of filters.

### CODE REVIEW

**Comment by richardbporter** [comment]
Do we want to axe the single event CSS/JS and theme it up in uids_base? There seems to be a problem with the current single event dates - some future dates are appearing as past.

**Comment by richardbporter** [comment]
Also looking for help on the heading level field and the LB styles for hiding image, description, etc. - can I reuse the ones created in #2032?

File: docroot/modules/custom/sitenow_events/sitenow_events.module [hunk 1]
```diff
@@ -6,169 +6,69 @@
...
- * @param int $parent
- *   Stores the current id.
- *
- * @return array
- *   Returns an associative array options tree.
- */
-function _build_options_tree(array $data, $parent = 0) {
-  $tree = [];
-  foreach ($data as $d) {
-    if ($d['parent_id'] == $parent) {
```
Comment (richardbporter) [inline docroot/modules/custom/sitenow_events/sitenow_events.module:132]: I think this loose comparison is a recursion trap but changing it breaks all the filters. It currently is equating null and 0.

**Comment by joewhitsitt** [comment]
>Do we want to axe the single event CSS/JS and theme it up in uids_base? There seems to be a problem with the current single event dates - some future dates are appearing as past.

Can you elaborate? I am not sure I follow.

>Also looking for help on the heading level field and the LB styles for hiding image, description, etc. - can I reuse the ones created in #2032?

I think that is a good idea. I plan to revisit the PR to work out the kinks and maybe reuse fields between the two blocks. Can this PR wait for that?

File: docroot/modules/custom/sitenow_events/sitenow_events.module (hunk 1 above)
Comment (richardbporter) [reply docroot/modules/custom/sitenow_events/sitenow_events.module:132]: Working on this.

**Comment by richardbporter** [comment]
>Can this PR wait for that?

Yes.

> Can you elaborate? I am not sure I follow.

Should we move the single-event.css file to the uids_base theme and possibly refactor? I feel like we could use an on-brand accordion for the dates dropdown instead of rolling our own.

**Comment by richardbporter** [comment]
Also. we could leverage some of the single event styles like location/date icons, etc.

**Comment by richardbporter** [comment]
I'm going to merge this branch into #2032 and close this PR so we can address a couple interconnected issues.

**Pull request closed without merging**